src/Thinktecture.Relay.Server/Middleware/RelayMiddleware.cs

```diff
@@ -318,32 +328,65 @@ private async Task<bool> TryInlineBodyContentAsync(TRequest request, Cancellatio
 		=> _responseWriter.WriteAsync(_relayContext.ClientRequest.CreateResponse<TResponse>(httpStatusCode), response,
 			cancellationToken);
 
-	private async Task<Guid?> LoadTenantByIdAsync(string name, CancellationToken cancellationToken = default)
+	private async Task<TenantInfo> LoadTenantInfoByNameAsync(string name, CancellationToken cancellationToken = default)
 	{
 		var normalizedName = _tenantService.NormalizeName(name);
 		var cacheKey = $"tenant_{normalizedName}_id";
 
-		var result = Guid.Empty;
+		var result = new TenantInfo();
 
-		var cachedId = await _cache.GetAsync(cacheKey, cancellationToken);
-		if (cachedId != null)
+		var cachedData = await _cache.GetAsync(cacheKey, cancellationToken);
+		if (cachedData != null)
 		{
-			result = new Guid(cachedId);
-			return result == Guid.Empty ? null : result;
+			return TenantInfo.FromByteArray(cachedData);
 		}
 
 		var tenant = await _tenantService.LoadTenantByNameAsync(name);
 		if (tenant != null)
 		{
-			result = tenant.Id;
+			result = new TenantInfo(tenant.Id, tenant.Connections?.Any(c => c.DisconnectTime == null) == true);
 		}
 
 		var cacheEntryOptions = new DistributedCacheEntryOptions()
-			.SetAbsoluteExpiration(_relayServerOptions.TenantIdCacheTimeout);
+			.SetAbsoluteExpiration(_relayServerOptions.TenantInformationCacheTimeout);
+
+		cachedData = result.ToByteArray();
+		await _cache.SetAsync(cacheKey, cachedData, cacheEntryOptions, cancellationToken);
+
+		return result;
+	}
+
+	private readonly struct TenantInfo
```

Comment:
Sometimes this is called `TenantInformation` and sometimes `TenantInfo`. This needs to be harmonized.